gravitee-am-test/specs/gateway/password-policy.jest.spec.ts

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from 'cross-fetch';
+import * as faker from 'faker';
+import { afterAll, afterEach, beforeAll, expect, jest } from '@jest/globals';
+import { requestAdminAccessToken } from '@management-commands/token-management-commands';
+import { createDomain, deleteDomain, startDomain } from '@management-commands/domain-management-commands';
+import { getIdp } from '@management-commands/idp-management-commands';
+import {
+  assignPasswordPolicyToIdp,
+  createPasswordPolicy,
+  deletePasswordPolicy,
+  getAllPasswordPolicies,
+  getPasswordPolicy,
+  resetUserPassword,
+  setPasswordPolicyDefault,
+} from '@management-commands/password-policy-management-commands';
+import { createTestApp } from '@utils-commands/application-commands';
+import { createMongoIdp } from '@utils-commands/idps-commands';
+import { createUser, deleteUser, getUser } from '@management-commands/user-management-commands';
+import { User } from '../../api/management/models';
+import { loginUserNameAndPassword } from '@gateway-commands/login-commands';
+import { getWellKnownOpenIdConfiguration, logoutUser } from '@gateway-commands/oauth-oidc-commands';
+
+global.fetch = fetch;
+jest.setTimeout(200000);
+let accessToken;
+let domain;
+let app;
+let customIdp;
+let customIdp2;
+let openIdConfiguration;
+let user;
+
+beforeAll(async () => {
+  const adminTokenResponse = await requestAdminAccessToken();
+  accessToken = adminTokenResponse.body.access_token;
+  expect(accessToken).toBeDefined();
+
+  const createdDomain = await createDomain(accessToken, 'domain-idp', faker.company.catchPhraseDescriptor());
+  expect(createdDomain).toBeDefined();
+  expect(createdDomain.id).toBeDefined();
+
+  const domainStarted = await startDomain(createdDomain.id, accessToken);
+  expect(domainStarted).toBeDefined();
+  expect(domainStarted.id).toEqual(createdDomain.id);
+
+  domain = domainStarted;
+
+  customIdp = await createMongoIdp(domain.id, accessToken);
+  customIdp2 = await createMongoIdp(domain.id, accessToken);
```

Comment:
todo: according to an env var, you have to create a Mongo or JDBC idp otherwise, the test will fail for PSQL integration as we start either Mongo or PSQL based on the job execution

Reply:
Ex here:

Env ver read : https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-test/specs/gateway/login-flow.jest.spec.ts#L29

IDP creation based on the env var : https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-test/specs/gateway/login-flow.jest.spec.ts#L45C80-L45C126

Reply:
fixed